use BUFSIZ as buffer size; fixes error in which BUFFERSIZE was not declared in scope

Fetched from: https://sources.debian.org/patches/libb64/1.2-5/

bufsiz-as-buffer-size.diff patch adapted for version 1.2.1 with
initialize-coder-state.diff already applied.

Signed-off-by: Robert B. Calhoun <rcalhoun@shotspotter.com>

Author: Jakub Wilk <jwilk@debian.org>

Bug: http://sourceforge.net/tracker/?func=detail&atid=785907&aid=3591336&group_id=152942

diff --git a/include/b64/decode.h b/include/b64/decode.h
index d3f7d60..fc7d248 100644
--- a/include/b64/decode.h
+++ b/include/b64/decode.h
@@ -8,6 +8,7 @@
 #ifndef BASE64_DECODE_H
 #define BASE64_DECODE_H

+#include <cstdio>
 #include <iostream>

 namespace base64
@@ -22,7 +23,7 @@ namespace base64
 		base64_decodestate _state;
 		int _buffersize;

-		decoder(int buffersize_in = BUFFERSIZE)
+		decoder(int buffersize_in = BUFSIZ)
 		: _buffersize(buffersize_in)
 		{
 			base64_init_decodestate(&_state);
diff --git a/include/b64/encode.h b/include/b64/encode.h
index 49aafdc..8c2d007 100644
--- a/include/b64/encode.h
+++ b/include/b64/encode.h
@@ -8,6 +8,7 @@
 #ifndef BASE64_ENCODE_H
 #define BASE64_ENCODE_H

+#include <cstdio>
 #include <iostream>

 namespace base64
@@ -22,7 +23,7 @@ namespace base64
 		base64_encodestate _state;
 		int _buffersize;

-		encoder(int buffersize_in = BUFFERSIZE)
+		encoder(int buffersize_in = BUFSIZ)
 		: _buffersize(buffersize_in)
 		{
 			base64_init_encodestate(&_state);
